src/include/duckdb/optimizer/cse_optimizer.hpp

```diff
@@ -30,7 +30,7 @@ class CommonSubExpressionOptimizer : public LogicalOperatorVisitor {
 	//! First iteration: count how many times each expression occurs
 	void CountExpressions(Expression &expr, CSEReplacementState &state);
 	//! Second iteration: perform the actual replacement of the duplicate expressions with common subexpressions nodes
-	void PerformCSEReplacement(unique_ptr<Expression> *expr, CSEReplacementState &state);
+	void PerformCSEReplacement(reference<unique_ptr<Expression>> expr, CSEReplacementState &state);
```

Comment:
We should be able to use `&` instead here, no?

Reply:
Hmm, actually I think you're right, it was just a raw pointer before, we couldn't reassign it anyways.
I'll change it to a reference 👍